src/store.rs

```diff
@@ -158,16 +161,25 @@ impl<'a> InnerStoreLocked<'a> {
                 .map_err(|_| err.to_string())
                 .and_then(|s| s.parse::<T>().map_err(|_| err.to_string()))
             {
-                Ok(value) => value,
+                Ok(v) => v,
                 Err(e) => return Err(e),
             },
-            None => T::default(),
+            None => T::zero(),
         };
 
         value += increment;
-        self.set(key.to_string(), value.to_string().into());
 
-        Ok(value)
+        let value = match value.to_f64() {
+            Some(v) if v.fract() == 0.0 => format!("{:.0}", v), // Format as an integer if no fractional part.
+            Some(v) => format!("{:.17}", v), // Format as a float with up to 17 digits of precision.
+            // This shouldn't happen since we're only using ints and floats, but ideally, a trait
+            // would enforce this at compile time.
+            None => return Err(err.to_string()),
+        };
```

Comment:
```suggestion
        let value = match value.to_f64() {
            // Format as an integer if no fractional part.
            Some(v) if v.fract() == 0.0 => format!("{:.0}", v),
            // Format as a float with up to 17 digits of precision.
            Some(v) => format!("{:.17}", v),
            // This shouldn't happen since we're only using ints and floats, but ideally, a trait
            // would enforce this at compile time.
            None => return Err(err.to_string()),
        };
```